includes/class-wc-stripe-account.php

```diff
@@ -81,6 +81,10 @@ private function cache_account() {
 			return [];
 		}
 
+		if ( is_wp_error( $account ) ) {
+			return [];
+		}
+
```

Comment:
@Mayisha , I like the idea of not caching an account that has errored. Looking at the `cache_account()` function here, it seems like we are trying to retrieve account info by calling our Stripe's client (`$this->stripe_api`). Then our Stripe client invokes a call to Stripe's endpoint at `https://api.stripe.com/v1/account` ([doc](https://stripe.com/docs/api/accounts/retrieve)).

To test, I updated my test key to an invalid key so that the account API will return an error. 

The API response will come back with a JSON error object. 
```
{
 error: {
    message: "...",
    type: "..."
 }
}
```
![image](https://user-images.githubusercontent.com/572862/134955231-6b1f09f2-7296-48c7-955f-8214d78951c4.png)

Now, when it goes into this `cache_account()` function, the `is_wp_error()` will pass because the Stripe error object is not a `WP_Error` object. 
![image](https://user-images.githubusercontent.com/572862/134955874-a3ea6fc0-507b-4a7f-ac00-766f884b4b64.png)

As a result, the Stripe error object is cached as `$account` and it will be used by code afterwards.

Should we check something like `if ( is_wp_error( $account ) && is_stripe_error($account))`?

Reply:
@harriswong nice catch :100: I missed this case previously. 

What do you think about handling this in the `retrieve` method itself? 

Change [this condition](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-api.php#L180) to the following and throw an exception? 
`if ( is_wp_error( $response ) || empty( $response['body'] ) || isset( $response['body']['message'] ) )`

Reply:
I think it is a good idea to check for errors in the `retrieve` function. Though, I would be a bit cautious when changing this behaviour. Currently, we return the `$response['body']` regardless of its HTTP statuses. We sometimes leave it to the function caller to deal with the error handling ([example](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/f5f9e87123a22c4c54e28f3a06e0c1af47de20be/includes/class-wc-stripe-customer.php#L300-L304)). If we throw an error before returning the Stripe's Error Object (`json_decode($response['body'])`), then existing error handling may be affected.

If we want to catch errors on the API response level, then I recommend we check all existing calls to `WC_Stripe_Customer::retrieve()` to make sure nothing breaks if a `WP_Error` is thrown.

> Change this condition to the following and throw an exception?
if ( is_wp_error( $response ) || empty( $response['body'] ) || isset( $response['body']['message'] ) )

The `$response['body']` is a JSON object from Stripe that contains more info on the error itself. To use it, you will need to do `json_decode()` first. Alternatively, we can also make use of the HTTP response code from `$response['code']`. We can probably throw errors for all non-200 responses, according to the [Stripe error doc](https://stripe.com/docs/api/errors).

Reply:
Does this block the 5.6 release? If so, I vote for patching the error notice, then refactoring the error handling in a follow-up issue.

Reply:
I have made the hotfix in [this commit](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1969/commits/d596c06711cd3363cf8c5b1002ffbe139320c4dd). It handles the case you mentioned in [this commit](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1969#discussion_r716895364).